packages/integration-karma/test/context/x/advancedProvider/advancedProvider.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * This is a referral implementation for a custom context provider
+ * that can be installed on any element in the dom, and can provide
+ * value to any LWC component using wire.
+ *
+ * This provide is sharing the same value with every child, and the
+ * identity of the consumer is not tracked.
+ */
+
+import { register, ValueChangedEvent, LinkContextEvent } from 'wire-service';
+
+const IdentityMetaMap = new WeakMap();
+const UniqueEventName = `advanced_context_event_${guid()}`;
+const Provider = Symbol('SimpleContextProvider');
+
+function guid() {
+    return Math.floor((1 + Math.random()) * 0x10000)
+        .toString(16)
+        .substring(1);
+}
+
+register(Provider, eventTarget => {
+    let unsubscribeCallback;
+
+    function callback(data, unsubscribe) {
+        eventTarget.dispatchEvent(new ValueChangedEvent(data));
+        unsubscribeCallback = unsubscribe;
+    }
+
+    eventTarget.addEventListener('connect', () => {
+        const event = new LinkContextEvent(UniqueEventName, callback);
+        eventTarget.dispatchEvent(event);
+        if (unsubscribeCallback === undefined) {
+            // no provider was found, in which case the default
+            // context should be set.
+            const defaultContext = null;
+            // Note: you might decide to use a global identity instead
+            eventTarget.dispatchEvent(new ValueChangedEvent(defaultContext));
+        }
+    });
+
+    eventTarget.addEventListener('disconnect', () => {
+        if (unsubscribeCallback !== undefined) {
+            unsubscribeCallback();
+            unsubscribeCallback = undefined; // resetting it to support reinsertion
+        }
+    });
+});
+
+function createNewConsumerMeta(provider, callback) {
+    // identity must be an object that can't be proxified otherwise we
+    // loose the identity when tracking the value.
+    const identity = Object.freeze(_ => {
```

Comment:
@jbenallen you were right, there were some issues with the identity of the object due to the fact that it becomes a proxy on the wired field setter. Using a function instead is enough.